File: nexacord-web/src/services/channelService.ts
import api from './api';
import type { Channel } from '../types';

class ChannelService {
  async getServerChannels(serverId: number): Promise<Channel[]> {
    return api.get<Channel[]>(`/channels/server/${serverId}`);
  }

  async getChannelById(channelId: number): Promise<Channel> {
    return api.get<Channel>(`/channels/${channelId}`);
  }

  async createChannel(
    serverId: number,
    channel: Pick<Channel, 'name' | 'type' | 'topic' | 'nsfw' | 'parentId'>
  ): Promise<Channel> {
    return api.post<Channel>(`/channels/server/${serverId}`, channel);
  }

  async updateChannel(channelId: number, channel: Partial<Channel>): Promise<Channel> {
    return api.put<Channel>(`/channels/${channelId}`, channel);
  }

  async deleteChannel(channelId: number): Promise<void> {
    return api.delete(`/channels/${channelId}`);
  }
}

export default new ChannelService();

File: nexacord-server/src/main/java/cn/cctstudio/nexacord/service/ChannelService.java
package cn.cctstudio.nexacord.service;

import cn.cctstudio.nexacord.model.Channel;
import cn.cctstudio.nexacord.model.Server;

import java.util.List;

public interface ChannelService {
    Channel createChannel(Channel channel, Server server);
    Channel getChannelById(Long id);
    List<Channel> getChannelsByServer(Server server);
    List<Channel> getChannelsByServerId(Long serverId);
    Channel updateChannel(Channel channel);
    void deleteChannel(Long id);
}